contest/views.py

```diff
@@ -76,63 +76,64 @@ def new(request):
-        raise Http404('Contest does not exist, can not edit.')
+    if request.user.is_authenticated():
+        try:
+            contest = Contest.objects.get(id = contest_id)
+        except Contest.DoesNotExist:
+            logger.warning('Contest: Can not edit contest %s! Contest not found!' % contest_id)
+            raise Http404('Contest does not exist, can not edit.')
 
-    if user_info.has_c_ownership(request.user,contest):
-        if request.method == 'GET':        
-            contest_dic = model_to_dict(contest)
-            form = ContestForm(initial = contest_dic)
-            return render(request,'contest/editContest.html',{'form':form,'user':request.user})
-        if request.method == 'POST':
-            form = ContestForm(request.POST, instance = contest)
-            if form.is_valid():
-                modified_contest = form.save()
-                logger.info('Contest: Modified contest %s!' % modified_contest.id)
-            return HttpResponseRedirect('/contest/')
-    else:
-        raise PermissionDenied
+        if user_info.has_c_ownership(request.user,contest):
+            if request.method == 'GET':        
+                contest_dic = model_to_dict(contest)
+                form = ContestForm(initial = contest_dic)
+                return render(request,'contest/editContest.html',{'form':form,'user':request.user})
+            if request.method == 'POST':
+                form = ContestForm(request.POST, instance = contest)
+                if form.is_valid():
+                    modified_contest = form.save()
+                    logger.info('Contest: Modified contest %s!' % modified_contest.id)
+                return HttpResponseRedirect('/contest/')
+    raise PermissionDenied
 
 def delete(request,contest_id):
-    try:
-        contest = Contest.objects.get(id = contest_id)
-    except Contest.DoesNotExist:
-        logger.warning('Contest: Can not delete contest %s! Contest not found!' % contest_id)
-        raise Http404('Contest does not exist, can not delete.')
-    
-    # only contest owner can delete
-    if request.user == contest.owner:
-        deleted_contest_id = contest.id
-        contest.delete()
-        logger.info('Contest: Delete contest %s!' % deleted_contest_id)
-        return HttpResponseRedirect('/contest/')
-    else:
-        raise PermissionDenied
+    if request.user.is_authenticated():
+        try:
+            contest = Contest.objects.get(id = contest_id)
+        except Contest.DoesNotExist:
+            logger.warning('Contest: Can not delete contest %s! Contest not found!' % contest_id)
+            raise Http404('Contest does not exist, can not delete.')
+        
+        # only contest owner can delete
+        if request.user == contest.owner:
+            deleted_contest_id = contest.id
+            contest.delete()
+            logger.info('Contest: Delete contest %s!' % deleted_contest_id)
+            return HttpResponseRedirect('/contest/')
```

Comment:
Better use ``reverse`` to avoid hard code, or you have to change all appearance if you modify ``url conf``
https://docs.djangoproject.com/en/1.7/topics/http/urls/#reverse-resolution-of-urls